test/e2e/parallel/user_workloads_test.go

```diff
@@ -71,12 +71,12 @@ func TestIdlerAndPriorityClass(t *testing.T) {
 	require.NoError(t, err)
 	_, err = memberAwait.WaitForPods(t, "workloads-noise", len(externalNsPodsNoise), wait.PodRunning(), wait.WithPodLabel("idler", "idler"), wait.WithOriginalPriorityClass())
 	require.NoError(t, err)
-	_, err = hostAwait.WithRetryOptions(wait.TimeoutOption(10*time.Second)).WaitForNotificationWithName(t, "test-idler-stage-idled", toolchainv1alpha1.NotificationTypeIdled, wait.UntilNotificationHasConditions(wait.Sent()))
-	require.True(t, errors.IsNotFound(err))
 
 	// Check if notification has been deleted before creating another pod
 	err = hostAwait.WaitUntilNotificationWithNameDeleted(t, "test-idler-dev-idled")
 	require.NoError(t, err)
+	err = hostAwait.WaitUntilNotificationWithNameDeleted(t, "test-idler-stage-idled")
```

Comment:
Are you checking that the notification for the stage idler was deleted or never created in the first place? The old code which you are fixing here is confusing indeed but what is actually supposed to happen here?

Reply:
It is to check that old notification for the stage idler was deleted

Reply:
But it looks like the stage idler is there just as noise and no notification for it is supposed to be created. If it's not the case then the comments above which say so are misleading.
For example this comment:
```
// make sure that "noise" pods are still there, and notification is not created for stage namespace
```

Reply:
I agree about the confusion, the question here would be, if its for noise, and expected to Not create any notification.., then why are we waiting for notification to be sent in the test case?
 
 ```
_, err = hostAwait.WithRetryOptions(wait.TimeoutOption(10*time.Second)).WaitForNotificationWithName(t, "test-idler-stage-idled", toolchainv1alpha1.NotificationTypeIdled, wait.UntilNotificationHasConditions(wait.Sent()))
	require.True(t, errors.IsNotFound(err))
```

This Test case will fail, because it will return nil (wont find the notification), as the notification is available just for 5 seconds and then deleted.

Reply:
ok, I missed that "noise" part during our call. 
@alexeykazakov is right - these lines were there for the very same reason as there are the lines:
https://github.com/codeready-toolchain/toolchain-e2e/blob/4a55451961e21fd546ba453a89fab787f07c880b/test/e2e/parallel/user_workloads_test.go#L91-L92

In other words, this is to check that the notification is not created at all - not deleted, but even not created. That's why we waited for 10 seconds to check that the notification is not really created.

Reply:
@fbm3307 what we could do is to improve the the code - there are two ways:
1. create a new function `WaitForNotificationToBeNotCreated` that will wait for 10 seconds and check that the Notification is not created - it will return an error if it is. 
2. Add additional `RetryOption` which could suppress the printed message:
```
hostAwait.WithRetryOptions(wait.TimeoutOption(10*time.Second), wait.SuppressDiff()).WaitForNotificationWithName(t, "test-idler-stage-idled", toolchainv1alpha1.NotificationTypeIdled, wait.UntilNotificationHasConditions(wait.Sent()))
```
so the `printNotificationWaitCriterionDiffs` method doesn't print anything.

TBH, I like the first option more since the function name says what we can expect to be executted

Reply:
how about i rename/update this function [https://github.com/codeready-toolchain/toolchain-e2e/pull/874/files#diff-2379a187b4778a2330f68b2b1a930fb26d3b8e1b567efe2966e4cb6d8aafe989R1214-R1226](url) instead of writing a new similar function (so that there is no redundancy)
My understanding is that the functionality of `WaitForNotificationToBeNotCreated `would be similar to `WaitUntilNotificationWithNameDeleted`, any thoughts around this..?

Reply:
and use that function to check that there are no notification created after 10 seconds

Reply:
i think writing a new func would be a better option

Reply:
The difference between waiting for something to be deleted and not created in the first place is the timeout. We would need a longer timeout for something which exist/existed but now we are waiting to be deleted. But we don't want to use the default (long) timeout for waiting for something which is not supposed to exist at all. This is why we usually wait no longer than 10 seconds in such case. So, we have to use different functions for those cases.

Reply:
If we always wait for the default timeout for something which doesn't exist then every successful test would wait for that long timeout. This is why a shorter timeout (10 seconds) is preferable.

Reply:
Thanks for explanation. I got a better Understanding now .